### PR TITLE
Add AssetScanner component

### LINKED ISSUE #39: New component AssetScanner
A React component from [OpSup](https://github.com/cognitedata/operations-support) to use the camera to scan&recognise an asset name (also called tag). Like 21PT1019. Using Google's Vision OCR.

Screenshot 2019-03-25 at 10 44 53

### PR DESCRIPTION
**Pull request by mrtraser**
Fixed #39

### CODE REVIEW

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-58.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
The AssetScanner doesn't work on the PR-storybook since the domain is not whitelisted...
Any way we can make the login optional? Or detect it is in a storybook someway?

**Comment by mrtraser** [comment]
We can make login process optional only if we will not use cognite-sdk in component (to get list of assets). What we can do is to wrap component in storybook with error handler (which will catch cross origin request error) and describe log in / register process in http://openindustrialdata.com for publicdata.

**Comment by f1cognite** [comment]
Yes, that is a good idea. But I don't know how you can catch it when it do a redirect?

One option is to render a pure component using the webcam, and when you hit scan it can show a popup explaining it requires you to be a user and use it locally or on a valid domain.

**Comment by mrtraser** [comment]
Facebook added Error Boundaries in React 16.
https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html

I was thinking about using such boundaries (cause I saw error in console in case of crossorigin error) to provide information about registration/login requirements

**Comment by f1cognite** [comment]
Yes, try that :-) It would be really good with a user-friendly message to the user

File: package.json [hunk 1]
```diff
@@ -23,12 +23,16 @@
...
-    "storybook": "start-storybook -p 6006"
+    "storybook": "start-storybook -p 6006",
+    "storybook-ssl": "start-storybook -p 6006 --https --ssl-cert ./server.cert --ssl-key ./server.key"
   },
   "repository": "cognitedata/gearbox.js",
   "homepage": "https://github.com/cognitedata/gearbox.js#readme",
   "license": "Apache-2.0",
   "dependencies": {
+    "@cognite/react-auth": "^2.1.2",
+    "@cognite/sdk": "^1.5.6",
```
Comment (f1cognite) [inline package.json]: This should be a peer-dependency (and a dev-dependency).

File: package.json [hunk 2]
```diff
@@ -23,12 +23,16 @@
...
     "test:watch": "yarn jest --watchAll",
-    "storybook": "start-storybook -p 6006"
+    "storybook": "start-storybook -p 6006",
+    "storybook-ssl": "start-storybook -p 6006 --https --ssl-cert ./server.cert --ssl-key ./server.key"
   },
   "repository": "cognitedata/gearbox.js",
   "homepage": "https://github.com/cognitedata/gearbox.js#readme",
   "license": "Apache-2.0",
   "dependencies": {
+    "@cognite/react-auth": "^2.1.2",
```
Comment (f1cognite) [inline package.json]: Dev-dependency? We should require the user of the AssetScanner to be already authenticated.

File: package.json [hunk 3]
```diff
@@ -23,12 +23,16 @@
...
+    "@cognite/sdk": "^1.5.6",
+    "axios": "^0.18.0",
```
Comment (f1cognite) [inline package.json]: What do we need axios for?

File: src/utils/api/api.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import axios from 'axios';
+import { Assets } from '@cognite/sdk';
+import { VAsset, VApiAssetList } from 'utils/validators';
+import { extractValidStrings } from 'utils/utils';
+
+const ocrURL = 'https://opin-api.cognite.ai/do_ocr';
```
Comment (f1cognite) [inline src/utils/api/api.ts]: Hm. I am unsure if we should rely on this non-public api-endpoint for doing OCR. Anything preventing us from doing the call to Google's Vision API directly? Exposing secret key?

File: package.json (hunk 2 above)
Comment (mrtraser) [reply package.json]: I'll remove this, we don't need it at all

File: package.json (hunk 3 above)
Comment (mrtraser) [reply package.json]: axios is using for handling async requests (for ocr request for example)

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-58.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-58.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-58.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-58.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-58.gearboxjs.preview.cogniteapp.com

**Pull request merged**